pkgs/development/coq-modules/tlc/default.nix

```diff
@@ -1,41 +1,24 @@
-{ stdenv, fetchurl, fetchFromGitHub, coq }:
+{ stdenv, lib, fetchurl, fetchFromGitHub, mkCoqDerivation, coq, version ? null }:
 
-let params =
-  if stdenv.lib.versionAtLeast coq.coq-version "8.10"
-  then rec {
-    version = "20200328";
-    src = fetchFromGitHub {
-      owner = "charguer";
-      repo = "tlc";
-      rev = version;
-      sha256 = "16vzild9gni8zhgb3qhmka47f8zagdh03k6nssif7drpim8233lx";
-    };
-  } else rec {
-    version = "20181116";
-    src = fetchurl {
-      url = "http://tlc.gforge.inria.fr/releases/tlc-${version}.tar.gz";
-      sha256 = "0iv6f6zmrv2lhq3xq57ipmw856ahsql754776ymv5wjm88ld63nm";
-    };
-  }
-; in
-
-stdenv.mkDerivation {
-  inherit (params) version src;
-  pname = "coq${coq.coq-version}-tlc";
-
-  buildInputs = [ coq ];
+with lib; mkCoqDerivation {
+  pname = "tlc";
+  owner = "charguer";
+  inherit version;
+  display-version = { tlc = false; };
+  default-version = with versions; switch coq.coq-version [
+    { case = is_ge "8.10"; out = "20200328"; }
+    { case = is_ge "8.6";  out = "20181116"; }
+  ] null;
+  release."20200328".sha256 = "16vzild9gni8zhgb3qhmka47f8zagdh03k6nssif7drpim8233lx";
+  release."20181116".sha256 = "0iv6f6zmrv2lhq3xq57ipmw856ahsql754776ymv5wjm88ld63nm";
+  release."20181116".domain = "tlc.gforge.inria.fr";
```

Comment:
If the tag was not reproduced on GitHub, can't you put the rev instead of changing the domain?

Reply:
I can certainly change the rev, could you find it?

Reply:
Actually, it's surprising but it looks like this tag does not exist either on the gforge repo.

Reply:
Anyway, there is a single commit at this date so this must be charguer/tlc@4babc16c2b37555f3f314bb9617a62ddbd347929. The `sha256` will probably require an update.

Reply:
What do you mean... `wget https://tlc.gforge.inria.fr/releases/tlc-20181116.tar.gz` succeeds

Reply:
maybe it's an "homemade" tarball ^^', should we ask Arthur ?

Reply:
This tarball does not correspond to any actual git tag. But I don't think there is much doubt on which commit this represents. If you want, we can open an issue on the GitHub repo to ask for this tag to be created.

Reply:
I think it would be desirable.

Reply:
Could you do it?

Reply:
Done. See https://github.com/charguer/tlc/issues/4